x/denommetadata/ibc_middleware.go

```diff
@@ -165,21 +159,10 @@ func (im IBCModule) OnAcknowledgementPacket(
 	}
 
 	state := im.hubKeeper.GetState(ctx)
-
-	// find the denom from the list by matching the base
-	// if it exists, update the status
-	// if the ack is error or timeout, update the status to INACTIVE
-	// otherwise, update the status to ACTIVE
-	for i, denom := range state.Hub.RegisteredDenoms {
-		if denom.Base == dm.Base {
-			state.Hub.RegisteredDenoms[i].Status = map[bool]hubtypes.RegisteredDenom_Status{
-				true:  hubtypes.RegisteredDenom_ACTIVE,
-				false: hubtypes.RegisteredDenom_INACTIVE,
-			}[ack.Success()]
-			im.hubKeeper.SetState(ctx, state)
-			break
-		}
-	}
+	state.Hub.RegisteredDenoms = append(state.Hub.RegisteredDenoms, &hubtypes.RegisteredDenom{
```

Comment:
we should only append if it doesnt exist. seems like we're appending duplicates.